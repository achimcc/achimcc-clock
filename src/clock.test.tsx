import React from 'react'
import ReactDOM from 'react-dom'
import Clock from './clock'
import {shallow} from 'enzyme'

describe('Clock', () => {
  it('is truthy', () => {
    expect(Clock).toBeTruthy()
  })
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Clock hours = {11} minutes = {33} />, div)
    ReactDOM.unmountComponentAtNode(div)
  })
  test('clock displays correct time', () => {
      const wrapper = shallow(<Clock hours = {11} minutes = {33} />)
      expect(wrapper.find('div').html()).toMatch(/11:33/) 
  })
  test('it should match snapshot', () => {
    const wrapper = shallow(<Clock hours = {11} minutes = {34} />)
    expect(wrapper).toMatchSnapshot()  
  })
})