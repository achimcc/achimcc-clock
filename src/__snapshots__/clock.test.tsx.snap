// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clock it should match snapshot 1`] = `
<div
  className="clock"
>
  11
  :
  34
</div>
`;
